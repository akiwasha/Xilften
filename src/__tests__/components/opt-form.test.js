import React from 'react';
import { render } from '@testing-library/react';
import { OptForm } from '../../components';

describe('<OptForm />', () => {
  it('renders the <OptForm /> with populated data', () => {
    const { container, getByText, getByPlaceholderText } = render(
      <OptForm>
        <OptForm.Input placeholder="Adresse e-mail" />
        <OptForm.Button>COMMENCER</OptForm.Button>
        <OptForm.Break />
        <OptForm.Text>
          Prêt à regarder xilften ? Saisissez votre adresse e-mail pour vous
          abonner ou réactiver votre abonnement.
        </OptForm.Text>
      </OptForm>
    );

    expect(getByText('COMMENCER')).toBeTruthy();
    expect(
      getByText(
        'Prêt à regarder xilften ? Saisissez votre adresse e-mail pour vous abonner ou réactiver votre abonnement.'
      )
    ).toBeTruthy();
    expect(getByPlaceholderText('Adresse e-mail')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
