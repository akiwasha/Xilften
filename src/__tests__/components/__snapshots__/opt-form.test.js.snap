// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptForm /> renders the <OptForm /> with populated data 1`] = `
<div
  class="opt-form__Container-sc-115zda8-0 jaBHSU"
>
  <input
    class="opt-form__Input-sc-115zda8-1 jpOxXZ"
    placeholder="Adresse e-mail"
  />
  <button
    class="opt-form__Button-sc-115zda8-2 igoxUK"
  >
    COMMENCER
     
    <img
      alt="Try Now"
      src="/images/icons/chevron-right.png"
    />
  </button>
  <div
    class="opt-form__Break-sc-115zda8-4 dUbazi"
  />
  <p
    class="opt-form__Text-sc-115zda8-3 gLFXgy"
  >
    Prêt à regarder xilften ? Saisissez votre adresse e-mail pour vous abonner ou réactiver votre abonnement.
  </p>
</div>
`;
